BrainPortal/lib/user_key_test_connection_error.rb

```diff
@@ -0,0 +1,32 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2012
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# This class provides an exception class for
+# representing a user key connection error.
+# At the moment it does not guaranty though that connection is good.
+# Currently # used on CbrainTask objects.
+class UserKeyTestConnectionError < CbrainException
```

Comment:
1. consider renaming (see previous comment)
2. even though you extracted the code form the NH codebase, you left the original class definition in the NH codebase too
3. Adjust the comment for the class
4. adjust the Copyright date range in the header (also do this with all other files you added)